src/middlewares/generateDocs.ts

```diff
@@ -0,0 +1,31 @@
+import path from 'path'
+import { logInfoMessage, logLineWithIcon, logErrorLine } from '../lib/logger.js'
+import Generator from '@asyncapi/generator'
+
+export default async (spec, config, resDir) => {
+  logInfoMessage(`Generating docs for your parsed specification...`)
+  const configData = config.generator
+  const resolvedData = spec._json
+  const generator = new Generator(
+    configData && configData.template
+      ? `@asyncapi/${configData.template}`
+      : '@asyncapi/markdown-template',
+    path.resolve(
+      resDir ? resDir : './',
+      configData && configData.folder ? configData.folder : 'docs'
+    )
+  )
+  try {
+    await generator.generateFromString(JSON.stringify(resolvedData))
+    logLineWithIcon(
+      ':zap:',
+      'Successfully generated docs for your specification'
```

Comment:
"specification" = https://github.com/asyncapi/spec/blob/master/spec/asyncapi.md
"document" = https://github.com/asyncapi/spec/blob/master/examples/simple.yml

People don't have specs but AsyncAPI documents. I'd keep it simpler and rephrase it as:

```suggestion
      'Successfully generated docs'
```